mutiny-core/src/nostr/mod.rs

```diff
@@ -173,6 +178,113 @@ impl<S: MutinyStorage> NostrManager<S> {
         Ok(profile)
     }
 
+    /// Lists all pending NWC invoices
+    pub fn get_pending_nwc_invoices(&self) -> Result<Vec<PendingNwcInvoice>, MutinyError> {
+        Ok(self
+            .storage
+            .get_data(PENDING_NWC_EVENTS_KEY)?
+            .unwrap_or_default())
+    }
+
+    /// Approves an invoice and sends the payment
+    pub async fn approve_invoice(
+        &self,
+        hash: sha256::Hash,
+        node_manager: &NodeManager<S>,
+        from_node: &PublicKey,
+    ) -> Result<EventId, MutinyError> {
+        let mut pending: Vec<PendingNwcInvoice> = self
+            .storage
+            .get_data(PENDING_NWC_EVENTS_KEY)?
+            .unwrap_or_default();
+
+        let inv = pending
+            .iter()
+            .find(|x| x.invoice.payment_hash() == &hash)
+            .ok_or(MutinyError::NotFound)?;
+
+        let nwc = {
+            let profiles = self.nwc.read().unwrap();
+            profiles
+                .iter()
+                .find(|x| x.profile.index == inv.index)
+                .ok_or(MutinyError::NotFound)?
+                .clone()
+        };
+
+        let client = Client::new(&self.primary_key);
+
+        #[cfg(target_arch = "wasm32")]
+        let add_relay_res = client.add_relay(&nwc.profile.relay).await;
+
+        #[cfg(not(target_arch = "wasm32"))]
+        let add_relay_res = client.add_relay(&nwc.profile.relay, None).await;
+
+        add_relay_res.expect("Failed to add relays");
+        client.connect().await;
+
+        let resp = nwc
+            .pay_nwc_invoice(node_manager, from_node, &inv.invoice)
+            .await;
+
+        let encrypted = encrypt(
+            &nwc.server_key.secret_key().unwrap(),
+            &nwc.client_pubkey(),
+            resp.as_json(),
+        )
+        .unwrap();
+
+        let p_tag = Tag::PubKey(inv.pubkey, None);
+        let e_tag = Tag::Event(inv.event_id, None, None);
+        let response = EventBuilder::new(Kind::WalletConnectResponse, encrypted, &[p_tag, e_tag])
+            .to_event(&nwc.server_key)
+            .unwrap();
```

Comment:
Unwrapping on nostr key stuff makes me nervous.

Reply:
The error would only happen if we didn't have the private key, changed to map the error tho